### PR TITLE
rpc/eth: query genesis to resolve earliest block

### PR DESCRIPTION
**Pull request by pro-wh**
make this block-number-to-round-parameter method more heavyweight: for "earliest," query the genesis block and use its round number.

### CODE REVIEW

File: rpc/eth/api.go [hunk 1]
```diff
@@ -568,14 +598,18 @@ func logs2EthLogs(logs []*Log, round uint64, blockHash, txHash common.Hash, txIn
 }
 
 func (api *PublicAPI) GetBlockHash(blockNum ethrpc.BlockNumber, _ bool) (common.Hash, error) {
-	return api.backend.QueryBlockHash(api.roundParamFromBlockNum(blockNum))
+	round, err := api.roundParamFromBlockNum(api.ctx, blockNum)
+	if err != nil {
+		return [32]byte{}, fmt.Errorf("convert block number to round: %w", err)
+	}
+	return api.backend.QueryBlockHash(round)
```
Comment (pro-wh) [inline rpc/eth/api.go:631]: and this one doesn't log errors \\:

**Comment by pro-wh** [comment]
oh and I messed with the error reporting. I'm really particular about that :tipping_hand_person:

**Comment by pro-wh** [comment]
changing base to main!

**Comment by pro-wh** [comment]
rebased, but did I just clobber this?

image


 1cec638 seems not to include this

File: rpc/eth/api.go [hunk 2]
```diff
@@ -394,7 +418,7 @@ func (api *PublicAPI) getRPCTransaction(dbTx *model.Transaction) (*utils.RPCTran
 	txIndex := hexutil.Uint64(dbTxRef.Index)
 	resTx, err := utils.NewRPCTransaction(dbTx, blockHash, dbTxRef.Round, txIndex)
 	if err != nil {
-		api.Logger.Error("Failed to NewRPCTransaction", "hash", dbTx.Hash, "err", err)
+		api.Logger.Error("Failed to NewRPCTransaction", "hash", dbTx.Hash, "error", err.Error())
```
Comment (pro-wh) [inline rpc/eth/api.go:446]: most existing code in this file used `"error", err.Error()`. I personally use `"err", err` in my own projects, but I'm changing these over for consistency throughout the file. leave a comment if these shouldn't be this way

**Comment by pro-wh** [comment]
> rebased, but did I just clobber this?

nvm, this was a message about the base branch

**Review by ptrus — APPROVED** [review]

**Pull request merged**